kornia/color/lab.py

```diff
@@ -0,0 +1,205 @@
+
+    xyz_im: torch.Tensor = rgb_to_xyz(image_s)
+
+    # normalize for D65 white point
+    xyz_ref_white: torch.Tensor = torch.tensor([0.95047, 1., 1.08883])[..., :, None, None]
+    xyz_ref_white = xyz_ref_white.to(xyz_im.device)
+    xyz_normalized = torch.div(xyz_im, xyz_ref_white)
+
+    xyz_int = torch.where(
+        xyz_normalized > 0.008856,
+        torch.pow(
+            xyz_normalized,
+            1 /
+            3),
+        7.787 *
+        xyz_normalized +
+        4. /
+        29.)
+
+    x: torch.Tensor = xyz_int[..., 0, :, :]
+    y: torch.Tensor = xyz_int[..., 1, :, :]
+    z: torch.Tensor = xyz_int[..., 2, :, :]
+
+    L = (116. * y) - 16.
+    a = 500. * (x - y)
+    b = 200. * (y - z)
+
+    out = torch.stack([L, a, b], dim=-3)
+
+    return out
+
+
+def lab_to_rgb(image: torch.Tensor, eps: float = 1e-12) -> torch.Tensor:
+    r"""Converts a Lab image to RGB.
+
+    Args:
+        image (torch.Tensor): Lab image to be converted to RGB with shape :math:`(*, 3, H, W)`.
+        eps (float): for numerically stability when dividing. Default: 1e-12.
+
+    Returns:
+        torch.Tensor: Lab version of the image with shape :math:`(*, 3, H, W)`.
+
+    Example:
+        >>> input = torch.rand(2, 3, 4, 5)
+        >>> output = lab_to_rgb(input)  # 2x3x4x5
+    """
+    if not isinstance(image, torch.Tensor):
+        raise TypeError("Input type is not a torch.Tensor. Got {}".format(
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
+                         .format(image.shape))
+
+    L: torch.Tensor = image[..., 0, :, :]
+    a: torch.Tensor = image[..., 1, :, :]
+    b: torch.Tensor = image[..., 2, :, :]
+
+    fy = (L + 16.) / 116.
+    fx = (a / 500.) + fy
+    fz = fy - (b / 200.)
+
+    fz = torch.where(fz < 0, torch.zeros_like(fz), fz)
+
+    fxyz = torch.stack([fx, fy, fz], dim=-3)
+
+    # Convert from Lab to XYZ
+    xyz: torch.Tensor = torch.where(fxyz > .2068966, torch.pow(fxyz, 3.0), (fxyz - 4. / 29.) / 7.787)
+
+    # For D65 white point
+    xyz_ref_white: torch.Tensor = torch.tensor([0.95047, 1., 1.08883])[..., :, None, None]
+    xyz_ref_white = xyz_ref_white.to(xyz.device)
+    xyz_im = xyz * xyz_ref_white
+
+    rgbs_im: torch.Tensor = xyz_to_rgb(xyz_im)
+
+    # Convert from sRGB to RGB Linear
+    rs: torch.Tensor = rgbs_im[..., 0, :, :]
+    gs: torch.Tensor = rgbs_im[..., 1, :, :]
+    bs: torch.Tensor = rgbs_im[..., 2, :, :]
+
+    r: torch.Tensor = torch.where(rs > 0.0031308, 1.055 * torch.pow(rs, 1 / 2.4) - 0.055, 12.92 * rs)
+    g: torch.Tensor = torch.where(gs > 0.0031308, 1.055 * torch.pow(gs, 1 / 2.4) - 0.055, 12.92 * gs)
+    b: torch.Tensor = torch.where(bs > 0.0031308, 1.055 * torch.pow(bs, 1 / 2.4) - 0.055, 12.92 * bs)
+
+    rgb_im: torch.Tensor = torch.stack([r, g, b], dim=-3)
+
+    # Clip to 0,1 https://www.w3.org/Graphics/Color/srgb
```

Comment:
[scikit-image](https://github.com/scikit-image/scikit-image/blob/master/skimage/color/colorconv.py#L618) does clipping, but if we do clip we won't have forth_and_back -> `rgb != lab2rgb(rgb2lab(rgb))` ... what to do?

Reply:
how does the output affect at all w/o the clipping ?

Reply:
I have tested what kind of values we get by using an image with all possible RGB values [0,255]. Opencv vs skimage vs kornia, all here in this gist
https://gist.github.com/cceyda/2a597b293782b07169328a92c7f7e979
The values that fall outside of [0,1] don't do so by much (~e-07). But since it effects the reversibility of the operation I think we can return the unclipped values (with a warning?) and leave it up to the user (like some other [libraries](https://python-colormath.readthedocs.io/en/latest/conversions.html#rgb-conversions-and-out-of-gamut-coordinates)) . OR do optional clipping with an clip=bool argument?

Other than that;
The current code produces the same results as skimage (without clipping).
There are some minor differences with opencv I'm assuming because of C vs python float op. differences. (Not sure if open cv does clipping)

Reply:
I wasn't aware of `python-colormath` - looks like a good image color mine :)
not sure raising a warning it's a good practice since the user using the functionality with the warning branch will receive annoying outputs every-time the function is called. Probably the `clip: bool` parameters is a better approach. /cc @ducha-aiki 
I believe other functionalities in this module can be adapted following this same convention.